libyul/AsmParser.cpp

```diff
@@ -154,6 +154,14 @@ void Parser::fetchDebugDataFromComment()
 	string_view commentLiteral = m_scanner->currentCommentLiteral();
 	match_results<string_view::const_iterator> match;
 
+	// (ugly) workaround to avoid regex seg fault with too large strings
+	size_t sizeLimit = 32 * 1024;
+	if (commentLiteral.size() > sizeLimit)
+	{
+		size_t resizeIndex = commentLiteral.find_first_of('\"');
+		commentLiteral = commentLiteral.substr(0, resizeIndex);
```

Comment:
Can we do that without information lost?

Reply:
Apparently, yes. 
Nevertheless, I don't think this should be the final solution. I thought about it a little and I guess it requires an alternative method to handle the special case of large strings that regex_search can't handle.

Reply:
Then, maybe we can just limit the string size in all cases without a need of differentiation between long and short strings?

Reply:
The problem is that sometimes you have more info (src location, ast id) after the code snippet and they need to be searched for and parsed. So, I implemented this workaround to only limit the string size if it would cause a seg fault.